src/main/scala/org/broadinstitute/hail/methods/ImputeSexPlink.scala

```diff
@@ -0,0 +1,129 @@
+package org.broadinstitute.hail.methods
+
+import java.nio.ByteBuffer
+
+import org.apache.spark.SparkEnv
+import org.apache.spark.rdd.RDD
+import org.broadinstitute.hail.Utils._
+import org.broadinstitute.hail.driver.ImputeSex.Options
+import org.broadinstitute.hail.variant.{VariantDataset, Variant, Genotype}
+import org.broadinstitute.hail.annotations._
+
+object ImputeSexPlink {
+  def imputeSex(vds: VariantDataset) = new ImputeSexPlink(vds)
+
+  def imputeSex(vds: VariantDataset, mafThreshold: Double, excludePAR: Boolean) =
+    new ImputeSexPlink(vds, mafThreshold, excludePAR)
+
+  def determineSex(ibc:InbreedingCombiner, fFemaleThreshold: Double, fMaleThreshold: Double): Option[Int] = {
+    ibc.F match {
+      case Some(x) =>
+        if (x < fFemaleThreshold)
+          Option(2)
+        else if (x > fMaleThreshold)
+          Option(1)
+        else
+          None
+      case None => None
+    }
+  }
+}
+
+class ImputeSexPlink(vds: VariantDataset, mafThreshold: Double = 0.0,
+                     excludePAR: Boolean = true) {
+
+  import ImputeSexPlink._
+
+  private def xChrVds =  {
+    if (excludePAR)
+      vds.filterVariants(
+      (v: Variant, va: Annotation, gs: Iterable[Genotype]) => !v.inParX && (v.contig == "X" || v.contig == "23")
+    )
+    else
+      vds.filterVariants(
+        (v: Variant, va: Annotation, gs: Iterable[Genotype]) => v.contig == "X" || v.contig == "23"
```

Comment:
I'd write this with one call to filterVariants: `(v.contig == "X" || v.contig == "23") && (!excludePAR || !v.inParX)`